draft-ietf-lamps-cms-kyber.md

```diff
@@ -27,474 +38,358 @@ author:
 
-In the following, *kekLen* denotes the length in bytes of the wrapping key for the underlying symmetric key-wrapping scheme.
-
-In this scheme, the length of the keying data to be transported MUST be among the lengths supported by the underlying symmetric key-wrapping scheme.
+~~~
+       (ct, ss) = Encapsulate(pk)
+~~~
 
-<!-- End of underlying-components section -->
+>
+2\. Derive a key-encryption key KEK from the shared secret:
 
-## Recipient's Key Generation and Distribution {#sec-key-generation-and-distribution}
+~~~
+       KEK = KDF(ss)
+~~~
 
-The KEM-TRANS described in the next sections assumes that the recipient has previously generated a key pair (*recipPrivKey* and *recipPubKey*) and has distributed his public key to the sender.
+>
+3\. Wrap the CEK with the KEK to obtain wrapped keying material WK:
 
-The protocols and mechanisms by which the key pair is securely generated and the public key is securely distributed are out of the scope of this document.
+~~~
+       WK = WRAP(KEK, CEK)
+~~~
 
-<!-- End of key-generation-and-distribution section -->
+>
+4\. The originator sends the ciphertext and WK to the recipient in the CMS KEMRecipientInfo structure.
 
-## Sender's Operations {#sec-sender-operations}
+To support the ML-KEM algorithm, the CMS recipient MUST implement Decapsulate().
 
-This process assumes that the following algorithm parameters have been selected:
+The ML-KEM algorithm recipient processing of the values obtained from the KEMRecipientInfo structure can be summarized as:
 
-- *KEM*: a key encapsulation mechanism, as defined above.
-- *KDF*: a key derivation function, as defined above.
-- *Wrap*: a symmetric key-wrapping algorithm, as defined above.
-- *kekLen*: the length in bits of the key required for the Wrap algorithm.
+>
+1\. Obtain the shared secret using the Decapsulate() function of the
+RSA-KEM algorithm and the recipient's ML-KEM private key:
 
-This process assumes that the following input data has been provided:
+~~~
+       ss = Decapsulate(sk, ct)
+~~~
 
-- *recipPubKey*: the recipient's public key.
-- *K*: the keying data to be transported, assumed to be a length that is compatible with the chosen Wrap algorithm.
+>
+2\. Derive a key-encryption key KEK from the shared secret:
 
-This process outputs:
+~~~
+       KEK = KDF(ss)
+~~~
 
-- *EK*: the encrypted keying data, from which the recipient will be able to retrieve *K*.
+>
+3\. Unwrap the WK with the KEK to obtain content-encryption key CEK:
 
+~~~
+       CEK = UNWRAP(KEK, WK)
+~~~
 
+Note that the KDF used to process the KEMRecipientInfo structure MAY be different from the KDF used in the ML-KEM algorithm.
 
-The sender performs the following operations:
+<!-- End of processing-summary section -->
 
-1. Generate a shared secret *SS* and the associated ciphertext *CT* using the KEM encaspulation function and the recipient's public key *recipPubKey*:
+## Conventions and Terminology {#sec-intro-terminology}
```

Comment:
This is odd to be so low in the document. Usually the boilerplate is section 1.1.

Reply:
Also copied from rfc5990bis, but I can move it up.